xdsl/interpreters/onnx.py

```diff
@@ -0,0 +1,82 @@
+import importlib.util
+from typing import Any, cast
+
+from xdsl.dialects import onnx
+from xdsl.interpreter import (
+    Interpreter,
+    InterpreterFunctions,
+    impl,
+    register_impls,
+)
+from xdsl.interpreters.shaped_array import ShapedArray
+
+where_is_numpy = importlib.util.find_spec("numpy")
+numpy_found = where_is_numpy is not None
+
+if numpy_found:
+    import numpy as np  # noqa: E402
```

Comment:
we now don't need this dance at all, the contract will be that everyone importing this file has to have checked the numpy dependency first. This is the case in this PR, where you test both in the unit test and rely on the flag in the `register_implementation` function